wearipedia/devices/fitbit/fitbit_sense_gen.py

```diff
@@ -0,0 +1,368 @@
+                                {
+                                    "caloriesOut": 0,
+                                    "max": 220,
+                                    "min": 169,
+                                    "minutes": 0,
+                                    "name": "Peak",
+                                },
+                            ],
+                            "restingHeartRate": 58,
+                        },
+                    }
+                ],
+                "activities-heart-intraday": {
+                    "dataset": [],
+                    "datasetInterval": 1,
+                    "datasetType": "minute",
+                },
+            }
+        ]
+    }
+
+    the_time = time(hour=0, minute=0, second=0)
+    for i in range(1440):
+
+        max_heart_rate = [110, 136, 169, 220]
+        min_dict = {110: 30, 136: 110, 169: 136, 220: 169}
+        weights = [0.14, 0.0058, 0.0008, 0.0001]
+        max_heart = choices(max_heart_rate, weights)
+
+        inx = max_heart_rate.index(max_heart[0])
+        heart_rate_data["heart_rate_day"][0]["activities-heart"][0]["value"][
+            "heartRateZones"
+        ][inx]["minutes"] += 1
+        heart_rate_data["heart_rate_day"][0]["activities-heart"][0]["value"][
+            "heartRateZones"
+        ][inx]["caloriesOut"] += 2 * (inx + 0.5)
+
+        val = np.random.randint(min_dict[max_heart[0]], max_heart)
+
+        heart_rate_data["heart_rate_day"][0]["activities-heart-intraday"][
+            "dataset"
+        ].append({"time": the_time.strftime("%H:%M:%S"), "value": val})
+
+        newtime = (
+            datetime.combine(datetime.today(), the_time) + timedelta(seconds=60)
+        ).time()
+        the_time = newtime
+
+    return heart_rate_data
+
+
+def get_hrv(date):
+    """Generate hrv for a given date.
+
+    :return: dictionary of hrv details
+    :rtype: dictionary
+    """
+    hrv = {
+        "hrv": [
+            {
+                "hrv": [
+                    {
+                        "value": {
+                            "dailyRmssd": np.random.randint(13, 48),
+                            "deepRmssd": np.random.randint(13, 48),
+                        },
+                        "dateTime": date,
+                    }
+                ]
+            }
+        ]
+    }
+
+    return hrv
+
+
+def get_distance_day(date):
+    """Generate distance data for a given date.
+
+    :return: dictionary of distance data details
+    :rtype: dictionary
+    """
+
+    distance_day = {
+        "distance_day": [
+            {
+                "activities-distance": [{"dateTime": date, "value": 0}],
+                "activities-distance-intraday": {
+                    "dataset": [],
+                    "datasetInterval": 1,
+                    "datasetType": "minute",
+                },
+            }
+        ]
+    }
+
+    the_time = time(hour=0, minute=0, second=0)
+
+    for i in range(1440):
```

Comment:
1440 as a magic number. Use a variable for duration instead, and name it so it is more readable.